koku/masu/database/ocp_report_db_accessor.py

```diff
@@ -823,6 +857,85 @@ def tag_upsert_monthly_node_cost_line_item(
                                     )
                                 line_item.save()
 
+    def tag_upsert_monthly_default_node_cost_line_item(
```

Comment:
I'd like to see if you can DRY these three upsert functions up a bit more.

It looks like, other than swapping out which `get_distinc_*` call they make, and identifying a couple of field names, these methods look like they are otherwise identical.

Let's see if we can make one generic function that we pass that information to, so that we can reduce the amount of duplicated code.

Reply:
Thats a good point and a good suggestion!

I do think the node and PVC ones could be combined into a more generic one, but the cluster one has a different structure due to the lack of a `get_distinct_*` for it so it doesn't have the same loop structure. I'll make a few changes!